documentation/example-file-usage/mqtt-example-file-usage.md

```diff
@@ -0,0 +1,136 @@
+
+3. Hold down the "Shift" key and right-click on some empty space in the repository top level folder. Select "Open in Terminal" from the context menu dropdown. (If the following commands do not work, try re-opening your terminal window as administrator and use the `cd` command to navigate to where you cloned this repository to)
+
+4. Enter in the following command to install/check Python 3:
+
+   ```powershell
+   python3
+   ```
+
+   Now, one of two things will happen: 
+
+   - The first possibility is that the Microsoft Store will open and prompt you to install Python 3.10. You can do that and that's a perfectly fine way to install Python 3 (it's the method I used) or you can go to the [Python website](https://www.python.org/downloads/) and install it from there. Either way works, do whichever you prefer. Enter in the above command again after and you should see what's in the next bullet.
+
+   - The other possibility is that you get a message similar to this:
+
+     ```
+     Python 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)] on win32
+     Type "help", "copyright", "credits" or "license" for more information.
+     >>> 
+     ```
+
+     This means you already have Python 3 installed and are good to go on to the next step. Enter in the following command to exit out of the Python 3 runtime environment.
+
+     ```python
+     quit()
+     ```
+
+5. Enter in the following command to install the required pip packages:
+
+   ```
+   pip3 install -r requirements.txt
+   ```
+
+   If required, enter `Y` to all installation prompts.
+
+### MacOS (Python 3, and pip packages)
+
+1. Install the AnkerMake slicer from [their website](https://www.ankermake.com/software). Make sure you open it and login, then you can close the app.
+
+2. Using `git` or GitHub Desktop, clone this repository into a location of your choice and then navigate to that location in Finder.
+
+3. Hold down the "Control" key and click the folder in the path bar, then choose "Open in Terminal". (If you don’t see the path bar at the bottom of the Finder window, choose View > Show Path Bar)
+
+4. Install Python3 from the [Python website.](https://www.python.org/downloads/macos/)
+
+5. Enter in the following command to install the required pip packages:
+
+   ```
+   pip3 install -r requirements.txt
+   ```
+
+   If required, enter `Y` to all installation prompts.
+
+
+
+## Extracting your Auth Token
+
+Your Auth Token is the "key to the kingdom" so-to-speak. The M5 uses this to verify who can connect to it over the internet (or LAN) to make sure it's actually you or someone you've shared your printer with (although, we haven't done any testing with shared printers so that's more of an educated guess). You're printer will refuse any connections that don't present your unique Auth Token, so let's get that out for future use.
```

Comment:
`s/You're/Your/`

And I have to clear up a technical misunderstanding here. The printer actually doesn't know about the `auth_token`! The token is used purely for a few endpoints in the https api, but we use that api to extract the mqtt key, and a few other key pieces of information.

Reply:
Ok, I definitely have a technical concept error here then. I'd love some amplifying info on that flow and maybe I can add that as a flow chart here instead.

Reply:
Sure thing!

When the printer is turned on, it will start sending regular mqtt updates to Ankermake mqtt servers. The script we're running here, will connect to those same servers, and subscribe to the same channels.

That way, we end up seeing the printer messages, but they're all bounced through Anker's servers. That's also how the slicer does it.

In fact, it's the only way to do it - both the printer and our script work as mqtt clients, so they need to connect to some mqtt server out there. Since the printer always connect to the ankermake servers, we do the same thing.

Reply:
And that's (indirectly) why we need the `auth_token`: to be able to connect to the https api, and request the mqtt login information.